tests/fixture/tmpnet/network.go

```diff
@@ -718,22 +774,20 @@ func (n *Network) CreateSubnets(ctx context.Context, w io.Writer) error {
 		}
 	}
 
-	if len(validatorsToRestart) == 0 {
-		return nil
-	}
-
-	if _, err := fmt.Fprintln(w, "Restarting node(s) to pick up chain configuration"); err != nil {
-		return err
-	}
-
-	// Restart nodes to allow configuration for the new chains to take effect
-	for _, node := range n.Nodes {
-		if !validatorsToRestart.Contains(node.NodeID) {
-			continue
-		}
-		if err := n.RestartNode(ctx, w, node); err != nil {
+	if restartRequired && len(validatorsToRestart) > 0 {
+		if _, err := fmt.Fprintln(w, "Restarting node(s) to pick up chain configuration"); err != nil {
 			return err
 		}
+
+		// Restart nodes to allow configuration for the new chains to take effect
+		for _, node := range n.Nodes {
+			if !validatorsToRestart.Contains(node.NodeID) {
+				continue
+			}
+			if err := n.RestartNode(ctx, w, node); err != nil {
+				return err
+			}
+		}
```

Comment:
Is there a reason we re-indented this chunk of code? It seems like we could have just modified the single if check to exit the function previously.

I think we generally try to avoid nesting to improve readability.